### ISSUE TITLE
Dropped shuttle causes a disconnect and soft locked save after it reached somewhere below 10000m.

### ISSUE THREAD

**Issue by StrayHALOMAN** [Regalis11/Barotrauma#9538]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

As the title says the server hard crashes everytime we play and the shuttle dropped from the Kestrul ship reaches 10000m "We didn't know it was dropped for a while"
Here are some screenshots and a small part of the console:

https://steamuserimages-a.akamaihd.net/ugc/1877454551960233470/66C33572EF0CF0458292A20BFC2FE7D55CEF5703/

![20220706212203_1](https://user-images.githubusercontent.com/108848191/177678344-2345d23c-7ded-411b-b255-a8efe1e5468b.jpg)

### Bug prevalence

Happens every time I play

### Version

0.18.12.0

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

```shell
Failed to read event 2365 for entity "Wifi Component (ID: 2765)", id 2765 {Index was outside the bounds of the array.}
at Lidgren.Network.NetBitWriter.ReadByte(Byte[] fromBuffer, Int32 numberOfBits, Int32 readBitOffset) in <DEV>\Libraries\Lidgren.Network\NetBitWriter.cs:line 41
at Lidgren.Network.NetBitWriter.ReadUInt16(Byte[] fromBuffer, Int32 numberOfBits, Int32 readBitOffset) in <DEV>\Libraries\Lidgren.Network\NetBitWriter.cs:line 229
at Barotrauma.Networking.MsgReader.ReadUInt16(Byte[] buf, Int32& bitPos) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Networking\Primitives\Message\Message.cs:line 261
at Barotrauma.Items.Components.ConnectionPanel.ApplyRemoteState(IReadMessage msg) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Items\Components\Signal\ConnectionPanel.cs:line 202
at Barotrauma.Items.Components.ConnectionPanel.ClientEventRead(IReadMessage msg, Single sendingTime) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Items\Components\Signal\ConnectionPanel.cs:line 173
at Barotrauma.Item.ClientEventRead(IReadMessage msg, Single sendingTime) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Items\Item.cs:line 1299
at Barotrauma.Networking.ClientEntityEventManager.Read(ServerNetObject type, IReadMessage msg, Single sendingTime, List`1 entities) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Networking\NetEntityEvent\ClientEntityEventManager.cs:line 224
```

**Comment by Regalis11** [comment]
Fixed in https://github.com/Regalis11/Barotrauma-development/commit/d686a264bfa008dc4e19d346f446dd8028864ea8

**Comment by StrayHALOMAN** [comment]
> Fixed in [Regalis11/Barotrauma-development@d686a26](https://github.com/Regalis11/Barotrauma-development/commit/d686a264bfa008dc4e19d346f446dd8028864ea8)
The link you sent isn't working for me.

**Comment by NilanthAnimosus** [comment]
Tested against dev commit https://github.com/Regalis11/Barotrauma-development/commit/3391dc7033d5d12b9167b38a6269ffd1096e546d, working correctly, closing.

also - the links are to the private github repository and more for the developers. the fix regalis linked above specifically being "Don't delete mapentities that end up below MaxEntityDepth if they belong to a sub because it leads to event errors"
